dhcpv4/ztpv4/parse_circuitid.go

```diff
@@ -0,0 +1,97 @@
+package ztpv4
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// CircuitID represents the structure of network vendor interface formats
+type CircuitID struct {
+	Slot		string
+	Module	string
+	Port		string
+	SubPort string
+	Vlan		string
+}
+
+var circuitRegexs = []*regexp.Regexp{
+	// Juniper QFX et-0/0/0:0.0
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+):(?P<subport>[0-9]+).*$"),
+	// Juniper PTX et-0/0/0.0
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+).*$"),
+	// Arista Ethernet3/17/1
+	// Juniper QFX et-1/0/61
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Arista Ethernet14:Vlan2001
+	// Arista Ethernet10:2020
+	regexp.MustCompile(".*Ethernet(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Cisco Gi1/10:2020
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Nexus Ethernet1/3
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper bundle interface ae52.0
+	regexp.MustCompile("^ae(?P<port>[0-9]+).(?P<subport>[0-9])$"),
+}
+
+// ParseCircuitId will parse dhcpv4 packet and return CircuitID info
+func ParseCircuitId(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
+	// CircuitId info is stored as sub-option field in RelayAgentInformationOption
+	relayOptions := packet.RelayAgentInfo() 
+
+	if relayOptions == nil {
+		return nil, fmt.Errorf("No relay agent information option found in the dhcpv4 pkt")
+	}
+	
+	if relayOptions.Options == nil {
+		return nil, fmt.Errorf("No relay agent information suboptions found in the dhcpv4 pkt")
+	}
+
+	// As per RFC 3046 sub-Option 1 is circuit-id. Look at 2.0 section in that RFC
+	// https://tools.ietf.org/html/rfc3046 
+	circuitIdStr := string(relayOptions.Options[1])
```

Comment:
To avoid doing `relayOptions.Options[1]`, I would define option codes for the relay agent information suboptions, similar to https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/bsdp/types.go#L30 but inside https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/option_relay_agent_information.go.

The sub options definitions are in https://www.iana.org/assignments/bootp-dhcp-parameters/bootp-dhcp-parameters.xhtml#relay-agent-sub-options

Then you can do something like `relayOptions.Options.Get(AgentCircuitIDSubOption)`.

Reply:
Hi Pablo, this make sense. I will add the sub-options in the option_relay_agent_information.go file. Thjis will increase the readability of the code. Thanks.